Modules/_hashopenssl.c

```diff
@@ -1833,13 +1834,128 @@ _hashlib_get_fips_mode_impl(PyObject *module)
 #endif  // !LIBRESSL_VERSION_NUMBER
 
 
+static int
+_tscmp(const unsigned char *a, const unsigned char *b,
+        Py_ssize_t len_a, Py_ssize_t len_b)
+{
+    /* loop count depends on length of b. Might leak very little timing
+     * information if sizes are different.
+     */
+    Py_ssize_t length = len_b;
+    const void *left = a;
+    const void *right = b;
+    int result = 0;
+
+    if (len_a != length) {
+        left = b;
+        result = 1;
+    }
+
+    result |= CRYPTO_memcmp(left, right, length);
+
+    return (result == 0);
+}
+
+/* NOTE: Keep in sync with _operator.c implementation. */
```

Comment:
if `configure.ac`'s $have_openssl block were expanded to expose a HAVE_OPENSSL_CRYPTO_MEMCMP as an AC_DEFINE, we could just make Modules/_operator.c conditionally use CRYPTO_memcmp?

Or would that wind up in linking hell as we'd need to link the main Python executable against ssl as Modules/Setup compiles _operator.c as a built-in?

Reply:
ah, saw your comment on the bug.

Reply:
The commit message also mentioned the fact that I don't want to link libcrypto into libpython.